primitive/bytes_map_test.go

```diff
@@ -102,11 +102,18 @@ func TestReadBytesMap(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			buf := bytes.NewBuffer(tt.source)
+			buf := bytes.NewReader(tt.source)
 			actual, err := ReadBytesMap(buf)
 			assert.Equal(t, tt.expected, actual)
-			assert.Equal(t, tt.remaining, buf.Bytes())
 			assert.Equal(t, tt.err, err)
+			remaining := make([]byte, buf.Len())
+			_, err = buf.Read(remaining)
+			if len(tt.remaining) == 0 {
+				assert.NotNil(t, err)
+			} else {
+				assert.Nil(t, err)
+				assert.Equal(t, tt.remaining, remaining)
+			}
```

Comment:
I think this and similar spots could be simplified:

```suggestion
			remaining, _ := ioutil.ReadAll(buf)
			assert.Equal(t, tt.remaining, remaining)
```

I think it's not necessary to capture the error from `ReadAll` since we know it cannot be non-nil.

Reply:
👍  done